recon/src/lib.rs

```diff
@@ -1,12 +1,14 @@
 //! Recon is a network protocol for set reconciliation
-#![deny(missing_docs)]
+#![warn(missing_docs, missing_debug_implementations, clippy::all)]
```

Comment:
Resolving so we can merge.
missing_debug_implementations was not being checked
 clippy::all was in the makefile but when I called build, I did not get the warnings.
 This way I can resolve the clippy warnings with context rather than as a pre push check